staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -4115,7 +4115,8 @@ const (
 // pod's hosts file.
 type HostAlias struct {
 	// IP address of the host file entry.
-	IP string `json:"ip,omitempty" protobuf:"bytes,1,opt,name=ip"`
+	// +required
+	IP string `json:"ip" protobuf:"bytes,1,opt,name=ip"`
```

Comment:
is it possible to leave `omitempty` in place?

Reply:
+1

Reply:
Leaving it in yields (in my tests):

```
The CustomResourceDefinition "dataplanes.gateway-operator.konghq.com" is invalid: spec.validation.openAPIV3Schema.properties[spec].properties[hostAliases].items.properties[ip].default: Required value: this property is in x-kubernetes-list-map-keys, so it must have a default or be a required property
```

With

```
type DataPlaneSpec struct {
...
	// HostAliases is an optional list of hosts and IPs that will be injected into the pod's hosts
	// file if specified.
	// +optional
	// +patchMergeKey=ip
	// +patchStrategy=merge
	// +listType=map
	// +listMapKey=ip
	HostAliases []HostAlias `json:"hostAliases,omitempty" patchStrategy:"merge" patchMergeKey:"ip" protobuf:"bytes,23,rep,name=hostAliases"`
}

type HostAlias struct {
	// IP address of the host file entry.
	// +required
	IP string `json:"ip" protobuf:"bytes,1,opt,name=ip"`
}
```

Reply:
see https://github.com/kubernetes/kubernetes/pull/124553#discussion_r1588479452

Reply:
since this field was actually required, I guess this had no effect on already-persisted data, but can we file an issue to teach kube-builder about `+required` as a follow-up and have it prefer that to an implicit omitempty, just like kube-openapi?

Reply:
https://github.com/kubernetes-sigs/kubebuilder/issues/3897